core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1034,6 +1034,14 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     logInfo("Added file " + path + " at " + key + " with timestamp " + addedFiles(key))
     postEnvironmentUpdate()
   }
+  
+  def supportKillExecutor(): Boolean = {
```

Comment:
I would call this `supportKillExecutor` and it should be made `private[spark]`.

Also can be shortened to just `private[spark] def supportsKillExecutor = master.contains("yarn") || dynamicAllocationTesting`.